UnitTests/UnitTests.UWP/UI/Controls/Test_WrapPanel_BasicLayout.cs

```diff
@@ -0,0 +1,340 @@
+                {
+                    var transform = treeRoot.CoordinatesTo(children[i]);
+                    Assert.AreEqual(expected[i].u, transform.X, $"Child {i} not in expected X location.");
+                    Assert.AreEqual(expected[i].v, transform.Y, $"Child {i} not in expected Y location.");
+
+                    Assert.AreEqual(100, children[i].ActualWidth, $"Child {i} not of expected width.");
+                    Assert.AreEqual(50, children[i].ActualHeight, $"Child {i} not of expected height.");
+                }
+            });
+        }
+
+        [TestCategory("WrapPanel")]
+        [TestMethod]
+        public async Task Test_WrapPanel_Normal_Horizontal_WithSpacing_AndPadding()
+        {
+            await App.DispatcherQueue.EnqueueAsync(async () =>
+            {
+                var treeRoot = XamlReader.Load(@"<Page
+    xmlns=""http://schemas.microsoft.com/winfx/2006/xaml/presentation""
+    xmlns:x=""http://schemas.microsoft.com/winfx/2006/xaml""
+    xmlns:controls=""using:Microsoft.Toolkit.Uwp.UI.Controls"">
+    <controls:WrapPanel x:Name=""WrapPanel"" HorizontalSpacing=""10"" Padding=""20"">
+        <Border Width=""100"" Height=""50""/>
+        <Border Width=""100"" Height=""50""/>
+        <Border Width=""100"" Height=""50""/>
+    </controls:WrapPanel>
+</Page>") as FrameworkElement;
+
+                var expected = new (int u, int v)[]
+                {
+                    (20, 20),
+                    (130, 20),
+                    (240, 20),
+                };
+
+                Assert.IsNotNull(treeRoot, "Could not load XAML tree.");
+
+                // Initialize Visual Tree
+                await SetTestContentAsync(treeRoot);
+
+                var panel = treeRoot.FindChild("WrapPanel") as WrapPanel;
+
+                Assert.IsNotNull(panel, "Could not find WrapPanel in tree.");
+
+                // Force Layout calculations
+                panel.UpdateLayout();
+
+                var children = panel.Children.Select(item => item as FrameworkElement).ToArray();
+
+                Assert.AreEqual(3, panel.Children.Count);
+
+                // Check all children are in expected places.
+                for (int i = 0; i < children.Count(); i++)
+                {
+                    var transform = treeRoot.CoordinatesTo(children[i]);
+                    Assert.AreEqual(expected[i].u, transform.X, $"Child {i} not in expected X location.");
+                    Assert.AreEqual(expected[i].v, transform.Y, $"Child {i} not in expected Y location.");
+
+                    Assert.AreEqual(100, children[i].ActualWidth, $"Child {i} not of expected width.");
+                    Assert.AreEqual(50, children[i].ActualHeight, $"Child {i} not of expected height.");
+                }
+            });
+        }
+
+        [TestCategory("WrapPanel")]
+        [TestMethod]
+        public async Task Test_WrapPanel_Normal_Vertical_WithSpacing_AndPadding()
```

Comment:
All the tests are doing the same things. I'm not using a lot _MSTest_ but there should be a way to have a single test and provide it the different inputs/expected results as test parameters (like using `Theory` in _xUnit_).

Reply:
The tests have different input XAML and expected outputs, if we had the same setup for each, I'd agree and have built with DataRow, but they're subtly different.